app/views/assessments/viewFeedback.html.erb

```diff
@@ -22,26 +22,41 @@ Yes, feedback is per problem but we show all annotations for this submission any
 <% if @submission.annotations.count > 0 and @problemReleased %>
   <h2 style="margin-top: 1em">Remarks</h2>
 
-  <ul>
-  <!-- use custom sort b/c using ordering doesn't work with autograder output, archived files -->
-  <% @subs = @submission.annotations.sort do |a,b|
-     aFilename = @get_correct_filename.call(a)
-     bFilename = @get_correct_filename.call(b)
-     case
-       when aFilename < bFilename
-         -1
-       when aFilename > bFilename
-          1
-       else
-          a.line <=> b.line
-       end
-     end
-     @subs.each do |annotation| %>
-    <li>
-      <%= link_to annotation.as_text, [:view, @course, @assessment, @submission, header_position: annotation.position] %>
-      <% filename = @get_correct_filename.call(annotation) %>
-      (<%= filename %>: <%= annotation.line %>)
-    </li>
-  <% end %>
-  </ul>
+  <% subs_by_problem = @submission.annotations.group_by(&:problem)
+      subs_by_problem.each do |problem, annotations| %>
+      <b> <%= problem.name %> </b>
+      <ul class="collection">
+      <!-- use custom sort b/c using ordering doesn't work with autograder output, archived files -->
+      <%
+        s_annotations = annotations.sort do |a,b|
+        aFilename = @get_correct_filename.call(a)
+        bFilename = @get_correct_filename.call(b)
+        case
+          when aFilename < bFilename
+            -1
+          when aFilename > bFilename
+              1
+          else
+              a.line <=> b.line
+          end
+        end
+        s_annotations.each do |annotation| %>
+        <li class="collection-item">
+              <div style='white-space: pre-wrap;'><%= annotation.comment%> </div>
+              <% filename = if annotation.position == -1
```

Comment:
Should use `get_correct_filename` helper for this

Reply:
Done